python/ray/data/_internal/split.py

```diff
@@ -218,7 +236,12 @@ def _split_at_indices(
     ] = _split_all_blocks(blocks_with_metadata, block_rows, per_block_split_indices)
 
     # phase 3: generate the final split.
-    return _generate_global_split_results(all_blocks_split_results)
+
+    # first calculate the size for each split.
+    helper = [0] + valid_indices + [sum(block_rows)]
```

Comment:
`valid_indices` are the user-provided indices without deduplication. The final blocks are generated based on `valid_indices`, so we will still generate empty blocks as before if user provides duplicated indices, FYI @jianoaix and @matthewdeng.